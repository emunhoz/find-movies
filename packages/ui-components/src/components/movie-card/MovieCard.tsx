import React from 'react'
import { IconHeartWhite, IconHeartFull, ImageNotFound } from '@commons/images'
import * as S from './MovieCard.style'

interface IMovieCard {
  imgUrl?: string
  title: string
  year: string
  onClick?: (e: any) => void
  active?: boolean
}

export const MovieCard = ({ active, imgUrl, title, year, onClick }: IMovieCard) => {
  const imgCard = imgUrl === 'N/A' ? ImageNotFound : imgUrl
  const activeImage = active ? IconHeartFull : IconHeartWhite

  return (
    <S.MovieCard>
      <S.Img loading="lazy" src={imgCard} alt={title} width="170" height="230" />
      <S.DetailLayer asFav={active!}>
        <S.IconWrapper>
          <img loading="lazy" src={activeImage} width="24" height="24" data-testid="set-as-fav" onClick={onClick}/>
        </S.IconWrapper>
        <S.DetailLayerWrapper>
          <S.Title>{title}</S.Title>
          <S.Year>{year}</S.Year>
        </S.DetailLayerWrapper>
      </S.DetailLayer>
    </S.MovieCard>
  )
}
