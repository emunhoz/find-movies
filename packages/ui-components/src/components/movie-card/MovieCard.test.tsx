import React from 'react'
import { render, fireEvent } from '../../utils/test-utils'
import { MovieCard } from './MovieCard'

describe('[Component]: MovieCard', () => {
  test('render the component', () => {
    const { asFragment } = render(<MovieCard imgUrl="N/A" title="Lorem" year="2020" />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('render the component active with image', () => {
    const { asFragment } = render(<MovieCard imgUrl="https://images-na.ssl-images-amazon.com/images/I/A1IuOyoadxL._AC_SY550_.jpg" title="Lorem" year="2020" />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('trigger function to set card as favourite', () => {
    const { getByTestId } = render(<MovieCard imgUrl="https://images-na.ssl-images-amazon.com/images/I/A1IuOyoadxL._AC_SY550_.jpg" title="My title" year="2020" />)
    const heartIcon = getByTestId('set-as-fav')

    fireEvent.click(heartIcon)

    expect(heartIcon).toBeVisible()
  })
})
