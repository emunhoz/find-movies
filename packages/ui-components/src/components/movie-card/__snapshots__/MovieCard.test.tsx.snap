// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component]: MovieCard render the component 1`] = `
<DocumentFragment>
  <div
    class="sc-iBPRYJ cXdPVu"
  >
    <img
      alt="Lorem"
      class="sc-dlfnbm CUPlz"
      src="test-file-stub"
    />
    <div
      class="sc-gKsewC pzKZi"
    >
      <div
        class="sc-hKgILt cfYQtU"
      >
        <img
          data-testid="set-as-fav"
          height="24"
          src="test-file-stub"
          width="24"
        />
      </div>
      <div
        class="sc-gsTCUz"
      >
        <h2
          class="sc-eCssSg drdQJI"
        >
          Lorem
        </h2>
        <div
          class="sc-jSgupP cPrTxs"
        >
          2020
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`[Component]: MovieCard render the component active with image 1`] = `
<DocumentFragment>
  <div
    class="sc-iBPRYJ cXdPVu"
  >
    <img
      alt="Lorem"
      class="sc-dlfnbm CUPlz"
      src="https://images-na.ssl-images-amazon.com/images/I/A1IuOyoadxL._AC_SY550_.jpg"
    />
    <div
      class="sc-gKsewC pzKZi"
    >
      <div
        class="sc-hKgILt cfYQtU"
      >
        <img
          data-testid="set-as-fav"
          height="24"
          src="test-file-stub"
          width="24"
        />
      </div>
      <div
        class="sc-gsTCUz"
      >
        <h2
          class="sc-eCssSg drdQJI"
        >
          Lorem
        </h2>
        <div
          class="sc-jSgupP cPrTxs"
        >
          2020
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
